src/main/java/dk/aau/p4/abaaja/Lib/Visitors/PrettyPrintVisitor.java

```diff
@@ -0,0 +1,338 @@
+        printIndented("return");
+        if(returnExp != null){
+            print(" ");
+            returnExp.accept(this);
+        }
+        print(";");
+        printNewline();
+    }
+
+    public void visit(FormalParamNode node){
+        print(node.get_id() + ": ");
+        node.get_type().accept(this);
+    }
+    public void visit(StopNode node){
+        printIndented("stop;");
+        printNewline();
+    }
+    public void visit(TypeNode node){
+        print(node.get_type());
+    }
+
+    public void visit(UnaryExpNode node){
+        ExpNode child = (ExpNode) node.get_children().get(0);
+
+        print(node.get_operatorLiteral());
+        if(child.get_arity() > 1) print("(");
+        child.accept(this);
+        if(child.get_arity() > 1) print(")");
+    }
+    public void visit(TypecastExpNode node){
+        ExpNode child = node.get_expression_node();
+
+        print("(");
+        node.get_typeNode().accept(this);
+        print(") ");
+        if(child.get_arity() > 1) print("(");
+        child.accept(this);
+        if(child.get_arity() > 1) print(")");
+    }
+    public void visit(BinaryExpNode node){
+        List<BaseNode> children = node.get_children();
+        ExpNode left = (ExpNode) children.get(0);
+        ExpNode right = (ExpNode) children.get(1);
+
+        if(left.get_arity() > 1) print("(");
+        left.accept(this);
+        if(left.get_arity() > 1) print(")");
+        print(" " + node.get_operatorLiteral() + " ");
+        if(right.get_arity() > 1) print("(");
+        right.accept(this);
+        if(right.get_arity() > 1) print(")");
+    }
+    public void visit(IDExpNode node){
+        // TODO: Why are these nodes not automatically visited by the correct visitor?
+        if(node instanceof IDArrayExpNode){
+            this.visit((IDArrayExpNode) node);
+            return;
+        }
+        if(node instanceof IDStructNode){
+            this.visit((IDStructNode) node);
+            return;
+        }
+        print(node.get_ID());
+    }
+    public void visit(IDArrayExpNode node){
+        node.get_IDNode().accept(this);
+        for (BaseNode child : node.get_children()) {
+            print("[");
+            child.accept(this);
+            print("]");
+        }
+    }
+    public void visit(IDStructNode node){
+        List<BaseNode> childList = node.get_children();
+        int numChildren = childList.size();
+
+        for (int i = 0; i < numChildren; i++) {
+            if(i > 0) print(".");
+            childList.get(i).accept(this);
+        }
+    }
+    public void visit(BoolExpNode node){
+        print(node.get_result() ? "true" : "false");
+    }
+    public void visit(NumExpNode node){
+        // The `replaceAll` removes unnecessary trailing zeroes from output
+        print(node.get_result().toString().replaceAll("\\.?0*(?=$)", ""));
+    }
+    public void visit(StringExpNode node){
+        print(node.get_result());
+    }
+
+    /**
+     * Useless crap
```

Comment:
Should this comment be changed? ❤️

Reply:
Sure, but it is true. You want it to be less spicy?

Reply:
Yes please

Reply:
Fixed in [c840cc6](https://github.com/AlbertHald/Minecraft-Turtle-Language/pull/17/commits/c840cc612bfa0cb4cca281e6b66944e323ad743c)